### PR TITLE
[typescript-angular2] access token function

### PR DESCRIPTION
**Pull request by chris1234p**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Allow the configuration property 'access_token' to be a string or a function, so that the value can be derived before each call, instead of just a single value that never changes. I was running into an issue where my access_token would expire and renew, but the value would not be set correctly while calling the API. Tested with my own angular2 project.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
cc @Vrolijkx

File: modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache [hunk 1]
```diff
@@ -143,7 +143,10 @@ export class {{classname}} {
         // oauth required
         if (this.configuration.accessToken)
         {
-            headers.set('Authorization', 'Bearer ' + this.configuration.accessToken);
+            let accessToken = typeof this.configuration.accessToken === 'function'
```
Comment (Vrolijkx) [inline modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache]: The type of configuration.accessToken in ```configuration.mustache``` should be ``` string | () => string```

Comment (chris1234p) [inline modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache]: @Vrolijkx I added a commit to address that

**Comment by wing328** [comment]
@Vrolijkx thanks for reviewing the change.

@chris1234p thanks for the PR, which has been merged into master.

**Comment by wing328** [comment]
(If your company is using Swagger Codegen, please consider adding your company name to [this list](https://github.com/swagger-api/swagger-codegen#companiesprojects-using-swagger-codegen))

**Comment by chris1234p** [comment]
@wing328 @Vrolijkx thanks - I was actually just testing something and found a small syntax error that will cause a warning. Should I issue a new PR or is there a way to add to this PR?

https://github.com/swagger-api/swagger-codegen/compare/master...chris1234p:typescript-angular2-access-token-function

**Comment by wing328** [comment]
@chris1234p yes, please submit a new PR for the enhancement.

**Pull request merged**